### PR TITLE
Updated, platform tool set to v143 (vs 2022), updated to c++20, improved code

### PR DESCRIPTION
**Pull request by Jasper1467**
[General]:
- Updated platform tool set to v143 (vs 2022)
- Updated to c++20
- Added consts
- Changed some NULL to nullptr when preferred
- Added [[maybe_unused]] attributes where preferred

Reversekit.cpp:
- Fixed bad usage of CreateThread, which could've led to race conditions, deadlocks, crashes and other unpredictable behavior

### CODE REVIEW

**Comment by mfxiaosheng** [comment]
Why have so many functions been marked as inline?

**Comment by Jasper1467** [comment]
> Why have so many functions been marked as inline?

In C++17 and later, inline keyword is no longer required in many cases, as the compiler can automatically decide whether to inline a function. However, in some cases where a function is defined in a header file and included in source files, the compiler may not be able to inline the function automatically. In these cases, adding the inline keyword to the function declaration can help the compiler to inline the function and improve performance.

So, in summary, inline is not required in most cases in modern C++, but can still be used as a hint to the compiler in cases where it may not be able to automatically inline the function.
